### PR TITLE
Fix STI for fixtures from multi-files

### PR DESCRIPTION
**Pull request by vipulnsward**
- Add check for not deleting previously created fixtures, to overcome sti fixtures from multiple files
- Added fixtures and fixtures test to verify the same

### CODE REVIEW

**Comment by vipulnsward** [comment]
Tests related to random named files are failing due to unique constraints.
I am not sure what the behaviour for them should be, I think the fixtures are wrongly duplicated.

File: activerecord/test/cases/fixtures_test.rb [hunk 1]
```diff
@@ -812,6 +813,13 @@ def test_supports_sti
     assert_equal pirates(:blackbeard), parrots(:polly).killer
   end
 
+  def test_supports_sti_with_respective_files
+    assert_kind_of LiveParrot, live_parrots(:dusty)
+    assert_kind_of DeadParrot, dead_parrots(:deadbird)
+    assert_equal pirates(:blackbeard), dead_parrots(:deadbird).killer
+  end
+
+
```
Comment (eileencodes) [inline activerecord/test/cases/fixtures_test.rb]: You added 2 spaces here, it should only be one :smile:

**Comment by vipulnsward** [comment]
Fixed spacing, also fixed random named fixture test, to not conflict data in a single table.

File: activerecord/lib/active_record/fixtures.rb [hunk 2]
```diff
@@ -521,12 +521,16 @@ def self.create_fixtures(fixtures_directory, fixture_set_names, class_names = {}
...
             fixture_sets.each do |fs|
               conn = fs.model_class.respond_to?(:connection) ? fs.model_class.connection : connection
               table_rows = fs.table_rows
 
               table_rows.each_key do |table|
-                conn.delete "DELETE FROM #{conn.quote_table_name(table)}", 'Fixture Delete'
+                unless deleted_tables.include? table
+                  conn.delete "DELETE FROM #{conn.quote_table_name(table)}", 'Fixture Delete'
+                end
+                deleted_tables << table
```
Comment (egilburg) [inline activerecord/lib/active_record/fixtures.rb]: Should this line be inside the `unless` block?

Comment (vipulnsward) [inline activerecord/lib/active_record/fixtures.rb]: either of the way works fine

**Comment by arthurnn** [comment]
backported https://github.com/rails/rails/commit/c79220368e8180141e9d796cf026d2b7da9f6384 to 4-2-stable. (https://github.com/rails/rails/commit/6a25fd1d694b8420fd6d525dd2bd243ea95ba490)

**Comment by arthurnn** [comment]
reverted this. https://github.com/rails/rails/commit/8097019c75a07eeda6025e138ef8073e717a3d25 , see the comment.

cc @jeremy

**Pull request merged**